### PR TITLE
CLN: Remove inheritance from object in pyx-files

### PR DESCRIPTION
**Pull request by topper-123**
- [x] xref #25725 and #26128

Continuation of #26128, this time dealing with pyx-files.

### CODE REVIEW

File: ci/code_checks.sh [hunk 1]
```diff
@@ -141,7 +141,7 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
     MSG='Check for python2 new-style classes' ; echo $MSG
-    invgrep -R --include="*.py" -E "class\s\S*\(object\):" pandas scripts
+    invgrep -R --include="*.py" --include="*.pyx" -E "class\s\S*\(object\):" pandas scripts
```
Comment (WillAyd) [inline ci/code_checks.sh]: Would make sense to just have one include flag like in the other PR

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Review by WillAyd — APPROVED** [review]
Lgtm

File: ci/code_checks.sh (hunk 1 above)
Comment (topper-123) [reply ci/code_checks.sh]: This follows the conventions of the code_checks.sh file. I also find it more readable than the other convention...

**Comment by gfyoung** [comment]
Thanks @topper-123 !

**Pull request merged**